### ISSUE TITLE
bug: Basic Examples run into NoneType-Issue in Starlette

### ISSUE THREAD

**Issue by Bernsai** [bentoml/OpenLLM#258]
### Describe the bug

The example code for remote access fails on my local machine and server, when I tried to get the basic OpenLLM example up and running with Llama2-13b-chat-hf and access it via the remote client. 

Interestingly, I can access the OpenAPI definition at <server-ip>:3000 and make calls manually (via curl or postman) to generate responses at POST <server-ip>:3000/v1/generate.

My current workaround is adapting method `iterate_in_threadpool` in `...\starlette\concurrency.py` to:

```python
async def iterate_in_threadpool(
    iterator: typing.Iterator[T],
) -> typing.AsyncIterator[T]:
    while True and iterator is not None:
        try:
            yield await anyio.to_thread.run_sync(_next, iterator)
        except _StopIteration:
            break
```

### On the server side:
1. `conda create --name openllm python=3.10`
2. `conda activate openllm`
3. `pip install openllm`
4. `openllm start llama --model_id meta-llama/Llama-2-13b-chat-hf`

### On the client side / my local machine:
1. `conda create --name openllm python=3.10`
2. `conda activate openllm`
3. `pip install openllm`
4. Execute the following script

```python
import openllm
client = openllm.client.HTTPClient('http://<server-ip>:3000')
response = client.query("I like to eat apples")
print(response)
```

### Logs

```shell
Traceback (most recent call last):
  File "c:\Users\<User>\dev\<openllm-project>\openllm_test.py", line 5, in <module>
    response = client.query("I like to eat apples")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\openllm\client\runtimes\base.py", line 127, in query
    prompt, generate_kwargs, postprocess_kwargs = self.llm.sanitize_parameters(prompt, use_default_prompt_template=use_default_prompt_template, **attrs)
                                                  ^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\openllm\client\runtimes\base.py", line 94, in llm
    if self.__llm__ is None: self.__llm__ = openllm.infer_auto_class(self.framework).for_model(self.model_name)
                                                                     ^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\openllm\client\runtimes\http.py", line 37, in framework
    try: return self._metadata["framework"]
                ^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\openllm\client\runtimes\base.py", line 62, in _metadata
    def _metadata(self) -> T: return httpx.post(urljoin(self._address, f"/{self._api_version}/metadata")).json() if in_async_context() else self.call("metadata")
                                                                                                                                            ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\openllm\client\runtimes\base.py", line 97, in call
    def call(self, name: str, *args: t.Any, **attrs: t.Any) -> T: return self._cached.call(f"{name}_{self._api_version}", *args, **attrs)
                                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\bentoml\_internal\client\__init__.py", line 53, in call
    return self._sync_call(
           ^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\bentoml\_internal\client\__init__.py", line 133, in _sync_call
    return asyncio.run(self._call(inp, _bentoml_api=_bentoml_api, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\bentoml\_internal\client\http.py", line 165, in _call
    req_body = "".join([s async for s in fake_resp.body_iterator])
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\bentoml\_internal\client\http.py", line 165, in <listcomp>
    req_body = "".join([s async for s in fake_resp.body_iterator])
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\starlette\concurrency.py", line 63, in iterate_in_threadpool
    yield await anyio.to_thread.run_sync(_next, iterator)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\<User>\.virtualenvs\<openllm-project>-...\Lib\site-packages\starlette\concurrency.py", line 53, in _next
    return next(iterator)
           ^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not an iterator
```

### Environment

transformers: 4.32.0
openllm: 0.2.26
Python  3.10.12
CuDNN 11.7

**Comment by aarnphm** [comment]
Can you upgrade to latest openllm verison? this should be fixed in 0.2.27

**Comment by Bernsai** [comment]
Thanks, a lot! Works great now :)
